ambry-network/src/main/java/com/github/ambry/network/FifoNetworkRequestQueue.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2022 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.network;
+
+import com.github.ambry.utils.Time;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A FIFO queue to hold network requests. It Internally, it maintains two queues, a bounded blocking queue to hold immediate
+ * set of next requests that can be handled by Ambry and unbounded queue to hold overflow requests.
```

Comment:
where is this unbounded queue ? is it requestsToDrop ? is it a good idea to have an unbounded queue ?